GPUVerifyLib/SourceLocationInfo.cs

```diff
@@ -222,6 +222,11 @@ public Record Top()
             return records[0];
         }
 
+        public List<Record> GetRecords()
+        {
+            return records;
+        }
+
```

Comment:
Why do you need this? You only seem to use this in `LocationChain.Equals`. The first half of that method re-implements `SourceLocationInfo.Equals`. For the second half it seems sufficient to expose `SourceLocationInfo.Top` and expose `SourceLocationInfo.Count`, or not?

Reply:
Note, by the way, that it's not guaranteed that there's more than one record, and if there's only one record that doesn't necessarily point to the entry point to the GPU kernel being verified. This depends on whether inlining in Bugle is turned on. It is by default. However, `LocationChain.Equals` seems to refers to one of our tests, and inlining is turned off for a lot test (for historical reasons).

Reply:
Apologies for the delayed reply. I was stuck with something else. I took a second look at this code and modified it to include the SourceLocation number in the class and removed the GetRecords method.
In GPURepair, we are reading the .loc file during initialization so that the source locations can be cached and we don't have to read the file multiple times. When inlining is enabled, each line in the .loc file represents multiple lines in the source code and we are using that information during the repair process. This information is stored in a class called [LocationChain](https://github.com/cs17resch01003/gpurepair/blob/main/src/Repair/Metadata/LocationChain.cs). This is similar to [SourceLocationInfo](https://github.com/cs17resch01003/gpuverify/blob/main/GPUVerifyLib/SourceLocationInfo.cs).

The primary use of GetRecords() was to map a SourceLocationInfo object to a LocationChain object. When I reviewed the code now, I realized that an easier way to do that was using the SourceLocation number since both these classes are referring to the same .loc file and the SourceLocation number represents the line number in the .loc file. Please let me know if this is okay to add the new property to the SourceLocationInfo class.